openedx/core/lib/cache_utils.py

```diff
@@ -6,40 +6,101 @@
 import functools
 import zlib
 
-from xblock.core import XBlock
+from django.utils.encoding import force_text
+from edx_django_utils.cache import RequestCache
 
 
-def memoize_in_request_cache(request_cache_attr_name=None):
+def request_cached(f):
     """
-    Memoize a method call's results in the request_cache if there's one. Creates the cache key by
-    joining the unicode of all the args with &; so, if your arg may use the default &, it may
-    have false hits.
+    A decorator for wrapping a function and automatically handles caching its return value, as well as returning
+    that cached value for subsequent calls to the same function, with the same parameters, within a given request.
+
+    Notes:
+        - We convert arguments and keyword arguments to their string form to build the cache key, so if you have
+          args/kwargs that can't be converted to strings, you're gonna have a bad time (don't do it)
+        - Cache key cardinality depends on the args/kwargs, so if you're caching a function that takes five arguments,
+          you might have deceptively low cache efficiency.  Prefer function with fewer arguments.
+        - We use the default request cache, not a named request cache. The code automatically namespaces the cache
+          key with the module and function's name.
+        - If you require a named request cache, use the ns_request_cached decorator below. Generally, you would need it
+          only if you need control of your own namespaced cache - for example, to clear out your own cache.
+        - WATCH OUT: Don't use this decorator for instance methods that take in a "self" argument that changes each
+          time the method is called. This will result in constant cache misses and not provide the performance benefit
+          you are looking for. Rather, change your instance method to a class method.
+        - Benchmark, benchmark, benchmark! if you never measure, how will you know you've improved? or regressed?
 
     Arguments:
-        request_cache_attr_name - The name of the field or property in this method's containing
-         class that stores the request_cache.
+        f (func): the function to wrap
+
+    Returns:
+        func: a wrapper function which will call the wrapped function, passing in the same args/kwargs,
+              cache the value it returns, and return that cached value for subsequent calls with the
+              same args/kwargs within a single request
+    """
+    return ns_request_cached()(f)
+
+
+def ns_request_cached(namespace=None, arg_map_function=None, request_cache_getter=None):
     """
-    def _decorator(func):
-        """Outer method decorator."""
-        @functools.wraps(func)
```

Comment:
Won't the name and docstring of the function that the decorator is applied to get wiped out without the `functools.wraps()` call?

Reply:
This is deleted code.  Do you see this issue in the new code as well?